python/educational-core/student/src/com/jetbrains/edu/learning/StudyTaskManager.java

```diff
@@ -114,7 +114,7 @@ public boolean hasFailedAnswerPlaceholders(@NotNull final TaskFile taskFile) {
   @Nullable
   @Override
   public Element getState() {
-    if (myCourse == null) {
+    if (myCourse == null && myUser.getEmail().isEmpty()) {
```

Comment:
This code is used not only in stepic integration but also with local courses. So user can be not logged in